pkg/volume/util/operationexecutor/operation_generator.go

```diff
@@ -276,7 +276,7 @@ func (og *operationGenerator) GenerateDetachVolumeFunc(
 			err = og.verifyVolumeIsSafeToDetach(volumeToDetach)
 		}
 		if err == nil {
-			err = volumeDetacher.Detach(volumeName, volumeToDetach.NodeName)
+			err = volumeDetacher.Detach(volumeToDetach.VolumeSpec, volumeName, volumeToDetach.NodeName)
```

Comment:
That is still the case. Which is why we left detach with only two parameters `volumeName` and `nodeName`. The full `volumeSpec` passed in via `volumeToDetach` is either a carry-over from the attach call (kept in-memory), or a "best-effort" reconstruction based on what was read of disk, and therefore fields other than the `volumeName` may not be correct.

@Quentin-M why is this change necessary? As far as I can tell, it's not used anywhere in your PR?

```GO
func (cdc *cdManagerCloud) Detach(_ *volume.Spec, ...)
```